src/System.Net.Security/src/System/Net/Unix/SSPISecureChannelType.cs

```diff
@@ -259,10 +260,10 @@ private SecurityStatus EncryptDecryptHelper(SafeDeleteContext securityContext, b
             {
                 securityContext.DangerousAddRef(ref gotReference);
                 inputHandle = GCHandle.Alloc(buffer, GCHandleType.Pinned);
-                IntPtr inputPtr = Marshal.UnsafeAddrOfPinnedArrayElement(buffer, offset);
+                IntPtr inputPtr = Marshal.UnsafeAddrOfPinnedArrayElement(buffer, 0);
                 if (encrypt)
                 {
-                    resultSize = Interop.OpenSsl.Encrypt(securityContext.DangerousGetHandle(), inputPtr, size, buffer.Length);
+                    resultSize = Interop.OpenSsl.Encrypt(securityContext.DangerousGetHandle(), inputPtr, offset, size, buffer.Length);
```

Comment:
> While you're touching this function, consider adding a Debug.Fail to catch to print out the exception that happens.

I'll leave that for @vijaykota and/or @rajansingh10.  I'd hope that the InternalError that it's returning would suggest that it's very unlikely; but wouldn't want to be responsible for the Debug.Fail that prints every time a remote host disconnects :).